arangod/VocBase/Methods/Collections.cpp

```diff
@@ -609,7 +609,15 @@ Result Collections::revisionId(
                                std::make_shared<VPackBuilder>(), arangodb::aql::PART_MAIN);
     auto queryRegistry = QueryRegistryFeature::QUERY_REGISTRY;
     TRI_ASSERT(queryRegistry != nullptr);
-    aql::QueryResult queryResult = query.execute(queryRegistry);
+    aql::QueryResult queryResult;
+    query.setContinueCallback([&query]() { query.tempSignalAsyncResponse(); });
+    while (true) {
+      auto state = query.execute(queryRegistry, queryResult);
+      if (state != aql::ExecutionState::WAITING) {
+        break;
+      }
+      query.tempWaitForAsyncResponse();
+    }
```

Comment:
This loop is repeated quite often. Maybe extract it into a query method that takes a QueryRegistry and returns a QueryResult?

Reply:
Yes you are right, will be done, before we move this into devel version, right now these two are intermediate branches. Thanks a lot for checking it!